pytorch_lightning/utilities/distributed.py

```diff
@@ -44,8 +44,18 @@ def wrapped_fn(*args, **kwargs):
     return wrapped_fn
 
 
+# TODO: this should be part of the cluster environment
+def _get_rank() -> int:
```

Comment:
Can you move this directly to SLURM cluster environment ?

What if `RANK`, `SLURM_PROCID` or `LOCAL_RANK` are different ? Should we take the latest or did you order `rank_keys` based on priority ?

Best,
T.C

Reply:
RANK = torchelastic
SLURM_PROCID = slurm
LOCAL_RANK = parity with existing profiler

I set local rank last as RANK and SLURM_PROCID correspond to global rank already. The linked issue has more discussion, but I think we should make global rank and world size properties of the cluster environment. So the cluster environment becomes the source of truth. Cluster environment => training type plugin => accelerator => trainer. 

If the cluster environment is marked as creating children, then we can leave the initialization of these fields for later, but both torchelastic and slurm have this data already available in the environment variables, and we should expose that as soon as possible (on Trainer init) for users to read this state.